R/calc_auc_analytic.R

```diff
@@ -0,0 +1,88 @@
+#' Convenience function to calculate the AUC based on PK model parameters at
+#' any given moment, for linear iv models.
+#'
+#' @param f analytic model to use, show available models using `advan()`
+#' @param regimen PKPDsim regimen created using `new_regimen`. Not required,
+#' regimen can also be specified using `dose`, `interval`, and `t_inf`.
+#' @param dose dosing amount. Only used if no `regimen` supplied.
+#' @param interval dosing interval. Only used if no `regimen` supplied.
+#' @param t_inf infusion length. Only used if no `regimen` supplied.
+#' @param parameters list of parameter estimates. Requires CL/V for
+#' 1-compartment models, CL/V/Q/V2 for 2-compartment models, and CL/V/Q/V2/Q2/V3
+#' for 3-compartment models.
+#' @param t_obs vector of observation times for AUC
+#' @param ... optional arguments passed to `advanc_create_data()`
+#'
+#' @returns a data.frame with `t` and `auc`
+#'
+#' @examples
+#' dat <- calc_auc_analytic(
+#'   f = "2cmt_iv_infusion",
+#'   regimen = new_regimen(
+#'     amt = 1000, n = 10, type = "infusion",
+#'     t_inf = 1, interval = 24
+#'   ),
+#'   parameters = list(CL = 5, V = 50, Q = 8, V2 = 150)
+#' )
+#'
+#' @export
+#'
+calc_auc_analytic <- function(
+    f = c(
+      "1cmt_iv_infusion", "2cmt_iv_infusion", "3cmt_iv_infusion",
+      "1cmt_iv_bolus", "2cmt_iv_bolus", "3cmt_iv_bolus"
+    ),
+    regimen = NULL,
+    dose = NULL,
+    interval = NULL,
+    t_inf = NULL,
+    parameters,
+    t_obs = c(0, 24, 48, 72),
+    ...
+) {
+  model <- match.arg(f)
+
+  if(is.null(regimen)) {
+    if(is.null(dose) || is.null(interval)) {
+      stop("Specify `regimen` or `dose` and `interval`")
+    }
+    if(is.null(t_inf) || t_inf == 0) t_inf <- 1e-6
```

Comment:
```suggestion
    if(is.null(t_inf) || any(t_inf == 0)) t_inf[t_inf > 0] <- 1e-6
```

t_inf can be of length > 1. the suggested code handles this situation.